core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1792,6 +1796,9 @@ class SparkContext(config: SparkConf) extends Logging {
 
   /**
    * Adds a JAR dependency for all tasks to be executed on this `SparkContext` in the future.
+   *
+   * If a jar is added during execution, it will not be available until the next TaskSet starts.
```

Comment:
This additional sentence is probably overkill, since I'm sure there are many other subtleties about these methods that aren't spelled out in the Javadoc.

I'm happy to remove and keep these docs simple if that's what a committer would prefer before merging.